src/lib/work_selector/work_lib.ml

```diff
@@ -87,9 +93,11 @@ module Make (Inputs : Intf.Inputs_intf) = struct
     unseen_jobs
 
   (*Seen/Unseen jobs that are not in the snark pool yet*)
-  let pending_work_statements ~snark_pool ~fee ~staged_ledger =
+  let pending_work_statements ~snark_pool ~fee_opt ~staged_ledger =
     let all_jobs = Inputs.Staged_ledger.all_work_pairs_exn staged_ledger in
     List.map
-      (get_expensive_work ~snark_pool ~fee all_jobs)
+      (Option.value_map fee_opt
+         ~default:(all_pending_work ~snark_pool all_jobs) ~f:(fun fee ->
```

Comment:
There's a downside to using `Option.value_map` in this way, which is that the `all_pending_work` call will be evaluated even if `fee_opt = Some _`. You can avoid this by either nesting lazy evaluations or by using a pattern match instead of an option combinator (or even defining a new combinator, but probably best to avoid that).

```ocaml
Lazy.force @@ Option.value_map fee_opt
  ~default:(lazy all_pending_work ~snark_pool all_jobs)
  ~f:(fun fee -> lazy get_expensive_work ~snark_pool ~fee all_jobs)
```

```ocaml
match fee_opt with
| None -> all_pending_work ~snark_pool all_jobs
| Some fee -> get_expensive_work ~snark_pool ~fee all_jobs
```